src/mbgl/programs/segment.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <mbgl/gl/context.hpp>
+#include <mbgl/gl/vertex_array.hpp>
+#include <mbgl/util/optional.hpp>
+
+#include <cstddef>
+#include <vector>
+#include <map>
+
+namespace mbgl {
+
+template <class Attributes>
```

Comment:
It's still useful as a means of type safety -- prevents mismatching a segment for the wrong set of attributes to `Program::draw`.